nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java

```diff
@@ -0,0 +1,172 @@
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.standard;
+
+import com.fluenda.parcefone.event.CEFHandlingException;
+import com.fluenda.parcefone.event.CommonEvent;
+import com.fluenda.parcefone.parser.CEFParser;
+
+
+import org.apache.nifi.annotation.behavior.EventDriven;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.behavior.InputRequirement.Requirement;
+import org.apache.nifi.annotation.behavior.SideEffectFree;
+import org.apache.nifi.annotation.behavior.SupportsBatching;
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.behavior.WritesAttributes;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.SeeAlso;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.InputStreamCallback;
+import org.apache.nifi.stream.io.StreamUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetAddress;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@EventDriven
+@SideEffectFree
+@SupportsBatching
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@Tags({"logs", "cef", "attributes", "system", "event", "message"})
+@CapabilityDescription("Parses the contents of a CEF formatted message and adds attributes to the FlowFile for " +
+        "headers and extensions of the parts of the CEF message.\n" +
+        "Note: This Processor expects CEF messages WITHOUT the syslog headers (i.e. starting at \"CEF:0\"")
+@WritesAttributes({@WritesAttribute(attribute = "cef.header.version", description = "The version of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceVendor", description = "The Device Vendor of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceProduct", description = "The deviceProduct of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceVersion", description = "The deviceVersion of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceEventClassId", description = "The deviceEventClassId of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.name", description = "The name of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.severity", description = "The severity of the CEF message."),
+    @WritesAttribute(attribute = "cef.extension.*", description = "The key and value generated by the parsing of the message.")})
```

Comment:
CEF messages are generally small but can get to around 100KB IIRC. Think of them being like syslog messages formatted in a particular way, reason why this processor is fairly similar to Bryan's ParseSyslog (where everything is an attribute). 

Having said that I am happy to offer an option of attribute vs json content as I suspect it will be handy for those using elasticsearch as final destination.